comp/process/agent/agentimpl/agent.go

```diff
@@ -0,0 +1,115 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+// Package agentimpl implements a component for the process agent.
+package agentimpl
+
+import (
+	"context"
+
+	"github.com/opentracing/opentracing-go/log"
+	"go.uber.org/fx"
+
+	"github.com/DataDog/datadog-agent/comp/core/config"
+	logComponent "github.com/DataDog/datadog-agent/comp/core/log"
+	"github.com/DataDog/datadog-agent/comp/process/agent"
+	"github.com/DataDog/datadog-agent/comp/process/runner"
+	"github.com/DataDog/datadog-agent/comp/process/types"
+	"github.com/DataDog/datadog-agent/pkg/process/checks"
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+	"github.com/DataDog/datadog-agent/pkg/util/optional"
+)
+
+const (
+	agentDisabledMessage = `process-agent not enabled.
+Set env var DD_PROCESS_CONFIG_PROCESS_COLLECTION_ENABLED=true or add
+process_config:
+  process_collection:
+    enabled: true
+to your datadog.yaml file.
+Exiting.`
+)
+
+// Module defines the fx options for this component.
+func Module() fxutil.Module {
+	return fxutil.Component(
+		fx.Provide(newProcessAgent))
+}
+
+type processAgentParams struct {
+	fx.In
+
+	Lc     fx.Lifecycle
+	Log    logComponent.Component
+	Config config.Component
+	Checks []types.CheckComponent `group:"check"`
+	Runner runner.Component
+}
+
+type processAgent struct {
+	Checks []checks.Check
+	Log    logComponent.Component
+	Runner runner.Component
+}
+
+func newProcessAgent(p processAgentParams) optional.Option[agent.Component] {
+	if !agentEnabled(p) {
+		return optional.NewNoneOption[agent.Component]()
+	}
+
+	// Look to see if any checks are enabled, if not, return since the agent doesn't need to be enabled.
+	if !checksEnabled(p.Checks) {
```

Comment:
Nit, optional: could this be combined with the populating of `enabledChecks` below?